grails-app/controllers/org/pih/warehouse/requisition/RequisitionTemplateController.groovy

```diff
@@ -86,6 +86,79 @@ class RequisitionTemplateController {
         }
     }
 
+    def exportPage = {
+        render(view:"exportPage")
+    }
+
+    def exportAsCsv = {
+        def requisitionItems = []
+        def origins = []
+        def destinations = []
+
+        params.origin.each { origin ->
+            origins << Location.get(origin)
+        }
+        params.destination.each { destination ->
+            destinations << Location.get(destination)
+        }
+
+        def requisitions = requisitionService.getRequisitionTemplates(origins, destinations)
+        requisitionService.getRequisitionTemplatesItems(requisitions).groupBy { it.product }.each { product, value ->
```

Comment:
You shouldn't need to do this unless you have a good reason. I would rather the two lines be combined into a single query:
```
getRequisitionTemplateItems(origins, destinations)
```

Reply:
Nevermind, I guess this is ok since you need the requisitions when generating the CSV. I thought you were doing this because you weren't sure how to write the query and wanted to decompose it into multiple queries.